src/Symfony/Component/DependencyInjection/Attribute/Autowire.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\DependencyInjection\Attribute;
+
+use Symfony\Component\DependencyInjection\Exception\LogicException;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\ExpressionLanguage\Expression;
+
+/**
+ * Attribute to tell a parameter how to be autowired.
+ *
+ * @author Kevin Bond <kevinbond@gmail.com>
+ */
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
```

Comment:
Do you think this `#[\Attribute(\Attribute::TARGET_PARAMETER, \Attribute::TARGET_PROPERTY)]` can be used here? When using Attribute with promoted property constructor.

Reply:
Yes you're right `TARGET_PARAMETER` is enough here. Thank you for your quick response :)

Reply:
I'm facing the same problem. 

`TARGET_PARAMETER` is working as well with properties for constructor, however not working especially with reflection mechanism on reflection property.

See https://3v4l.org/gDeYA what I mean.

Also PHPStorm suggests not to use this attribute on property promoted:
![obraz](https://user-images.githubusercontent.com/16488888/174674603-a1f31f4b-da90-467a-aacd-83147edd3861.png)

cc @kbond @nicolas-grekas @qdequippe

Reply:
Could this be a bug in PHP or a missing detail in the implementation? The opposite is a problem too: https://3v4l.org/tBjj2. Somewhat related PHPStan issue: https://github.com/phpstan/phpstan/issues/4418.

Reply:
As far as I understand, PHPStan literally "silence"  this situation, right?

For me isn't looks like a PHP bug, in my opinion we should add TARGET_PROPERTY flag. ;)

Reply:
To me it's a limitation of the engine and the engine should be improved: from a descriptive pov, both variant give all the possible hints to allow making things clear. The engine is not clever enough yet.
The alternative is to NOT use CPP. Patching the attribute would be wrong: Autowire should NOT be allowed on properties.
Could you please open a bug report about this on php-src?

Reply:
> See https://3v4l.org/gDeYA what I mean.

Calling `newInstance()` for arbitrary attributes is a footgun. Attributes could point to undefined classes and the attribute constructor could have side effects. The reflection API by design provides your with enough information to decide whether you need an actual instance of an attribute. Only call `newInstance()` on attributes you know and that your code actually needs to work with.

I agree with @nicolas-grekas that there's nothing we should do on our side.

Reply:
I agree as well. At least until a time where the attribute can actually be used with [property injection](https://symfony.com/doc/current/service_container/injection_types.html#property-injection).